ddcz/forms/authentication.py

```diff
@@ -38,3 +40,197 @@ def get_users(self, email):
+                "id": "motive",
+            }
+        ),
+    )
+
+    kamaradi_na_webu = forms.CharField(
+        required=False,
+        label="",
+        widget=forms.Textarea(
+            attrs={
+                "placeholder": "Znáš ve městě někoho mimo Virga?",
+                "id": "web_friends",
+                "class": "form-control",
+            }
+        ),
+    )
+
+    odkud_znas_web = forms.CharField(
+        required=False,
+        label="",
+        widget=forms.Textarea(
+            attrs={
+                "placeholder": "Odkud ses dozvěděl o existenci tohoto města (webu)?",
+                "id": "source",
+            }
+        ),
+    )
+
+    gdpr = forms.ChoiceField(
+        label="",
+        widget=forms.Select(attrs={"id": "gdpr"}),
+        choices=GDPR_CHOICES.items(),
+    )
+
+    class Meta:
+        model = UzivateleCekajici
+        fields = [
+            "nick_uzivatele",
+            "email",
+            "jmeno",
+            "prijmeni",
+            "pohlavi",
+            "osloveni",
+            "datum",
+            "patron",
+            "primluvy",
+            "popis_text",
+        ]
+
+    # Setting the data
+    @classmethod
+    def set_for_save(cls, data):
+        data["datum"] = int(datetime.now().timestamp())
+
+        data["primluvy"] = 0
+        data["patron"] = 0
+        data["popis_text"] = cls.set_description(data)
+        return data
+
+    @staticmethod
+    def set_description(data):
+        popis = "1. " + data["duvod_registrace"] + "\n"
+        popis += "2. " + data["odkud_znas_web"] + "\n"
+        popis += "3. " + data["kamaradi_na_webu"] + "\n"
+        popis += "4. \n"
+        popis += "5. \n"
+        return popis
+
+    # Validations for the form
+    def clean_nick_uzivatele(self, *args, **kwargs):
+        nick_uzivatele = self.cleaned_data.get("nick_uzivatele")
+        bad_characters = []
+        for char in self.FORBIDDEN_NICK_CHARACTERS:
+            if char in nick_uzivatele:
+                bad_characters.append(char)
+        if bad_characters.__len__() > 0:
+            raise forms.ValidationError(
+                "Ve vašem nicku jsou některé ze zakázaných znaků: ("
+                + ",".join(bad_characters)
+                + ")."
+            )
+        return nick_uzivatele
+
+    def clean_vek(self, *args, **kwargs):
+        vek = self.cleaned_data.get("vek")
+        if vek < self.MIN_AGE:
+            raise forms.ValidationError(
+                "Bohužel ti není " + self.MIN_AGE + " let. Zkus to, až budeš starší."
+            )
+        return vek
+
+    def clean_pohlavi(self, *args, **kwargs):
+        pohlavi = self.cleaned_data.get("pohlavi")
+        if pohlavi not in ["0", "1"]:
+            raise forms.ValidationError(
+                "Bohužel toto pohlaví nám zůstává utajeno. Neplete si pohlaví s genderem?"
```

Comment:
I'd skip on this joke since given how we use it, we are actually asking about the gender and not sex (is used in the language). 

Reflecting on that, It would maybe be better to ask "Jak vás máme oslovovat?". I think it fits better with the page anyway: we don't give a flying fuck about whether the actual sex/gender/identity is real in the real-world sense and this leaves up to interpretation whether people want to provide real identity or assumed identity, in either gender or role-playing way.

Reply:
Re-reading the story, I think it's fine the way it is. I'd still skip on the joke though and say something along the lines of `Ale nějakou gramatiku použít musíme` :)

Reply:
Shall we include all? :D https://www.healthline.com/health/different-genders
Although we could be a little progressive and try to add some more then just male/female if you're really talking about genders.

This actually shouldn't have been a joke, I just wasn't sure what to write as an error line. It will only occurs if someone messes up with the select, it can't happen.

I believe we can also skip this into the next version of registration, since I believe it need to be thought more thoroughly.

Reply:
What you posted are not genders :P But again—I'd sidestep the issue as I don't think we even need to store it beside the "how to you want to be grammatically called". Fair point about the next gen.